apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/Components/MainStatSelectionCard.tsx

```diff
@@ -84,132 +84,143 @@ export default function MainStatSelectionCard({
       slotTots: catTotals.slotTots,
     }
   }, [database, filteredArtIdMap])
-
-  return (
-    <Box display="flex" flexDirection="column">
-      <Divider />
-      <Box display="flex">
-        <CardContent sx={{ flexGrow: 1 }}>
-          <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
-            <BootstrapTooltip
-              placement="top"
-              title={<Typography>{t(`slotName.flower`)}</Typography>}
-            >
-              <FlowerIcon fontSize="inherit" />
-            </BootstrapTooltip>
-            <Box flexGrow={1}>
-              <SqBadge color="info">
-                <HpIcon {...iconInlineProps} /> {mainStatSlotTots.flower.hp}
-              </SqBadge>
-            </Box>
-          </Box>
-        </CardContent>
-        <Divider orientation="vertical" flexItem />
-        <CardContent sx={{ flexGrow: 1 }}>
-          <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
-            <BootstrapTooltip
-              placement="top"
-              title={<Typography>{t(`slotName.plume`)}</Typography>}
-            >
-              <PlumeIcon fontSize="inherit" />
-            </BootstrapTooltip>
-            <Box flexGrow={1}>
-              <SqBadge color="info">
-                <AtkIcon {...iconInlineProps} /> {mainStatSlotTots.plume.atk}
-              </SqBadge>
+  return useMemo(
```

Comment:
I think the way to go is to convert this into a `React.memo` function, and propdrill in `mainStatKeys`, and use `arePropsEqual` to check for update.